src/Config.php

```diff
@@ -23,9 +25,16 @@ public function getAllowSourceProjectId(): ?string
         return $this->getImageParameters()['sourceProjectId'] ?? null;
     }
 
+    /**
+     * @throws \Keboola\Component\UserException
+     */
     public function getToken(): string
     {
-        return $this->getImageParameters()['#token'] ?? $this->getValue(['parameters', '#token']);
+        try {
+            return $this->getImageParameters()['#token'] ?? $this->getValue(['parameters', '#token']);
+        } catch (InvalidArgumentException $e) {
+            throw new UserException('Token has to be set in config or image parameters.');
```

Comment:
uživatel moc neví co je Image params... nepsal bych to tam a napsal bych tam tu generickou hlášku že `parameters.#token` není vyplněný